amplify/backend/api/colonycdapp/schema.graphql

```diff
@@ -501,4 +513,99 @@ type ColonyAction @model {
   toDomain: Domain @hasOne(fields: ["toDomainId"])
   fundamentalChainId: Int
   newColonyVersion: Int
+  roles: ColonyActionRoles
+
+  # Required since some actions might have multiple event entries in the list
+  # which need to be iterated over and exposed individually
+  #
+  # This should be a list/array, but this is yet another thing Amplify is bad at
+  # The official response from them is to use DynamoDB directly:
+  # https://stackoverflow.com/questions/50992106/append-item-to-list-using-aws-appsync-to-dynamodb
+  # Which is just freaking great!
+  #
+  # As such, I've decided to simplify this, an just pass a stringified JSON around
+  # since this is for display purpouses only.
+  #
+  # If anyone has a better idea on how to handle this, I'll gladly change it!
+  #
+  # When / if you decide to use it, just be mindful of what format other people
+  # are using, and follow suit.
+  #
+  # Roles Types:
+  # [{ id: String, type: String, role: Number, setTo: boolean }]
+  individualEvents: String
+}
+
+type ColonyActionRoles {
+  role_0: Boolean # recovery
+  role_1: Boolean # root
+  role_2: Boolean # arbitration
+  role_3: Boolean # architecture
+  role_5: Boolean # funding
+  role_6: Boolean # administration
+}
+
+# Roles have been set as role_<number> so we can more easily map them to their
+# actual contract counterpart on the client and block ingestor side(s)
+
+type ColonyRole @model {
+  id: ID! @index # colonyAddress_domainNativeId_userAddress_roles
+  domain: Domain! @hasOne
+
+  targetAddress: ID # Amplify will automatically populate one of the following fields with related model if it finds one
+  targetUser: User @hasOne(fields: ["targetAddress"])
+  targetColony: Colony @hasOne(fields: ["targetAddress"])
+  targetExtension: ColonyExtension @hasOne(fields: ["targetAddress"])
+  targetToken: Token @hasOne(fields: ["targetAddress"])
+
+  latestBlock: Int!
+  role_0: Boolean # recovery
+  role_1: Boolean # root
+  role_2: Boolean # arbitration
+  role_3: Boolean # architecture
+  role_5: Boolean # funding
+  role_6: Boolean # administration
+}
+
+# it's a model, not a type since this is designed to work outside of the main
+# colony model, or colony role model, as, by design it should be only used
+# in very specific cases
```

Comment:
I don't think I know enough but why does it need to work outside of the colony role model?

Reply:
Outside, as it not directly linked (via @hasOne @hasMany, etc) as this is envisioned to just be fetched on an instance by instance basis. 

This will become relevant for Recovery Mode.